graylog2-web-interface/src/components/inputs/InputStateControl.jsx

```diff
@@ -0,0 +1,48 @@
+import React, {PropTypes} from 'react';
+import Reflux from 'reflux';
+import { Button } from 'react-bootstrap';
+
+import { Spinner } from 'components/common';
+
+import { InputStatesStore } from 'stores/inputs';
+
+function inputStateFilter(state) {
+  return state.inputStates[this.props.input.id];
+}
+
+const InputStateControl = React.createClass({
+  propTypes: {
+    input: PropTypes.object.isRequired,
+  },
+  mixins: [Reflux.connectFilter(InputStatesStore, 'inputState', inputStateFilter)],
+  _isInputRunning() {
+    if (!this.state.inputState) {
+      return false;
+    }
+
+    const nodeIDs = Object.keys(this.state.inputState);
+    let isInputRunning = true;
+
+    for (let i = 0; i < nodeIDs.length && isInputRunning; i++) {
```

Comment:
I haven't seen any case where `nodeIDs.length` was 0, it certainly didn't happen when stopping inputs on local and global inputs on my tests. The response from the server includes information for each node, even if the information is `null` or an empty array.

Anyway, you are right that it may lead to some weird state if that ever happen, so I'll take a quick look into it.